### ISSUE TITLE
opt: support CREATE VIEW

### ISSUE THREAD

**Issue by RaduBerinde** [cockroachdb/cockroach#38940]
CREATE VIEW (and more recently CREATE TABLE AS) relies on the heuristic planner to fully qualify datasource names. That means we can't use queries that are not supported by the HP with views, and will be a blocker to removing the HP altogether.

```
root@127.93.220.160:41221/defaultdb> select a,b from ab where exists (select x from xy where x=a);
  a | b  
+---+---+
(0 rows)

Time: 2.699971ms

root@127.93.220.160:41221/defaultdb> create view v as select a,b from ab where exists (select x from xy where x=a);
pq: column "a" does not exist
```

**Comment by Gurio** [comment]
Hi @RaduBerinde I would like to pick up some more or less "real" task, and do it
But I am very new to cockroachdb, so I would need time and guidance.
Do you think I can take this one? (it is fine if not, it might be urgent for example)

**Comment by RaduBerinde** [comment]
Hi Arseni,

Thank you but unfortunately this is not a good first issue; it involves some subtelties in both the legacy heuristic planner code and the new optimizer code. I'm familiar with both and can't say I have a clear idea how we will solve it :)

**Comment by awoods187** [comment]
@Gurio you can try an issue from our good first issue label https://github.com/cockroachdb/cockroach/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22 and someone will be more than happy to assist you

**Comment by Gurio** [comment]
@RaduBerinde @awoods187 OK, no problem, and thanks for the answer!
I've already done one recent entry-level task, but there are not much of those, tbh :)
Most of those are more than half a year old, and I am a bit reluctant to pick such a task.
For example this one: https://github.com/cockroachdb/cockroach/issues/36439 seem to be resolved by two PRs #37174 and #38513 
Anyway, you are right, there are some other entry-level tasks that seems I can pick up.
